### ISSUE TITLE
Feature Request: shift click to select multiple items - summarizes total gold value

### ISSUE THREAD

**Issue by Rezy942** [Auctionator/Journalator#43]
Hello, first of all thank you for amazing addon.
The biggest reason I need this is to gauge how good of a profit I made buying/selling on AH.
Currently, I dont see an option to easily summarize selected fields. I have to add one after another.
Feature request: add some functionality to select and summarize multiple items (easily) to see total gold value. Similar to spreadsheets.

Use case scenario example:
I buy 10 different reagents on AH to craft an item.
I sell said item for profit/loss.
I then go to Journalator to summarize total gold spent buying reagents, then look at profit I made by selling the item.

To go even further (perhaps unnecessary) it could be a 2-step function where you first summarize the items bought on AH, then summarize the items sold on AH, and addon spits out the difference. (gold gained - gold lost).

Hoping for your consideration
Thank you!

**Comment by plusmouse** [comment]
You can enter multiple items into the search field, separated by |
![image](https://user-images.githubusercontent.com/60280559/218992657-4a839363-61d2-403f-be76-c9c98f198a68.png)

**Comment by Rezy942** [comment]
Yes I am aware but that is extreme amount of work as you have to record what items you used (sometimes as many as 10+ reagents in a single craft) then go and enter those. If you click on item only 1 gets entered in search. Rest you have to go off memory.

Another issue is multiple purchases of same items for different crafts. That would get summarized all together (you can select within last hour, but again if you did that in the last hour it gets confusing)

Basically what im asking for is a quick way to summarize gold value of specific entries without entering another tab - straight from the main window. 
Would this be possible?

Alternatively hold and click to select items - which inputs ALL selected item names into that search bar you mentioned.
It wouldnt fix the "multiple crafts" issue I mentioned, but its a great QoL improvement.

**Comment by plusmouse** [comment]
Any thoughts on using tracked recipes to select the recipe reagents and results to search for?

**Comment by Rezy942** [comment]
Could be a decent idea but probably more complicated - this recipe is a good example.
When you go to track recipe in AH to buy reagents via shopping list - rimefin tuna is not counted because you have to buy it under different name and "open" the fishes.
So possible this could be a first problem right there..

**Comment by Rezy942** [comment]
An elegant solution (in my eyes) would be to allow user to right click items which simply selects it. (currently no actions for right click)
User can select as many rows as they want.
Then at bottom right it would spit out the difference between total gained and total lost.

Quick example:
![image](https://user-images.githubusercontent.com/122007087/219023151-f5427fcf-8db5-4833-bbea-a17ad017d605.png)

Obvious if user only selects negative values, the addon would spit out "- x gold" and + if only positive.

**Comment by fatmagic** [comment]
Ooo what a great feature this would be to select the rows you want to tally! Looking forward to this in a future full release :)
+1 for my vote!

**Comment by Rezy942** [comment]
Thank you so much!!!! Will test it out soon :)

**Comment by Rezy942** [comment]
Hi - are you still planning to add this feature?
What you made worked pretty well besides some bugs (which I assume should be easy to fix)
Thanks

**Comment by plusmouse** [comment]
Hi. I've spent some time on it today, got a much better version 😄 [Journalator-1.32-6-ge679371.zip](https://github.com/Auctionator/Journalator/files/11242583/Journalator-1.32-6-ge679371.zip)

![image](https://user-images.githubusercontent.com/60280559/232321977-c197ae57-079f-4f7c-a530-a4b823a881af.png)

You can select positive and negative amounts, and from any tab.

**Comment by Rezy942** [comment]
Thank you SO much!!!
Any chance this can get added to the main addon?
Reason being I want to be able to update the addon in the future :)

**Comment by Rezy942** [comment]
Just tested this - see screenshot:
![image](https://user-images.githubusercontent.com/122007087/235283728-9b5bd1a2-9f38-4124-9065-d8ea37dba888.png)

**Comment by plusmouse** [comment]
Fixed, find it in the latest alpha, available here or from Curseforge/Wago shortly.
[Journalator-1.35-1-g99d7764.zip](https://github.com/Auctionator/Journalator/files/11358679/Journalator-1.35-1-g99d7764.zip)
![image](https://user-images.githubusercontent.com/60280559/235292254-2fff99a2-ae45-4bc3-93ee-5645f7f3abaa.png)

**Comment by Rezy942** [comment]
Youre amazing, thank you.
